code/controllers/subsystem/dcs.dm

```diff
@@ -57,4 +85,16 @@ PROCESSING_SUBSYSTEM_DEF(dcs)
 		named_arguments = sortTim(named_arguments, /proc/cmp_text_asc)
 		fullid += named_arguments
 
-	return list2params(fullid)
+	var/id = list2params(fullid)
+
+	if (isnull(cached_id_info))
+		if (isnull(ids_per_type[eletype]))
+			ids_per_type[eletype] = list((source_path) = list(arguments.Copy(), id))
+		else
+			ids_per_type[eletype][source_path] = list(arguments.Copy(), id)
```

Comment:
what happens if arguments contains a datum. will it hang a ref?

Reply:
Yes, I think we're going to find out with create and destroy if anything does that. Though I can't think of a valid element that would do that?

Reply:
a bespoke element could pass in a second object to register to, and increment the cache index.
That's what comes directly to mind. feels silly to break that behavior

Reply:
I'll do a scan through of all new elements before putting into the cache

Reply:
would it be possible to only save the arguments that are bespoke?
then we can just count backwards through both lists
we just don't get the best case optimization as a side effect, and potentially hang refs to bespoke args

Reply:
I don't know what you mean by that sorry

Reply:
@LemonInTheDark What do you mean by that

Reply:
I don't like that we can currently save non bespoke args in our cache list
I would prefer to instead just the bespoke arguments, and then walk both lists backwards to compare them.
this means the length comparison either requires a cut, or we need to discard it and walk the full list each time.

I could be wrong about that tho, think I am. but that's what I was thinking when I said it

Reply:
I don't see how walking backwards is going to help anything if the lengths are equivalent